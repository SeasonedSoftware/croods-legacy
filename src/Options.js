import { Component } from 'react'

import withOptions from './withOptions'

class Options extends Component {
  render() {
    const { render } = this.props

    return render(this.props)
  }
}

export default withOptions(Options)
