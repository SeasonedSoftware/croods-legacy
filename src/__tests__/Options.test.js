import React from 'react'
import renderer from 'react-test-renderer'

import Options from '../Options'

jest.mock('../Context', () => ({
  Consumer: ({ children }) => (
    <div>Consumer - {children({ foo: 'bar', quu: 'foo' })}</div>
  ),
}))


it('renders correctly', () => {
  const props = { foo: 'bar' }
  const render = props => <div {...props}>Render</div>
  const tree = renderer.create(<Options {...props} render={render} />).toJSON()
  expect(tree).toMatchSnapshot()
})
