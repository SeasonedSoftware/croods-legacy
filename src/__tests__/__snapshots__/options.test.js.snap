// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Consumer - 
  <div
    foo="bar"
    quu="foo"
    render={[Function]}
  >
    Render
  </div>
</div>
`;
